src/optionals/log.c

```diff
@@ -0,0 +1,302 @@
+#include <string.h>
+#include <time.h>
+
+#include "log.h"
+#include "optionals.h"
+#include "../vm/vm.h"
+
+typedef struct {
+    FILE *of;
+    char *prefix;
+} Log;
+
+#define AS_LOG(v) ((Log*)AS_ABSTRACT(v)->data)
+
+static struct tm* timeNow() {
+    time_t rawtime = time(NULL);
+    if (rawtime == -1) {
+        return NULL;
+    }
+    
+    struct tm *ptm = localtime(&rawtime);
+    if (ptm == NULL) { 
+        return NULL;
+    }
+
+    return ptm;
+}
+
+static Value printLog(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 1) {
+        runtimeError(vm, "print() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    struct tm *ptm = timeNow();
+
+    char *msg = AS_CSTRING(args[0]);
+    printf("%04d/%02d/%02d %02d:%02d:%02d %s", 
+        ptm->tm_year+2000-100, ptm->tm_mon, ptm->tm_mday, 
+        ptm->tm_hour, ptm->tm_min, ptm->tm_sec, msg);
+
+    return newResultSuccess(vm, NIL_VAL);
```

Comment:
This can't have a recoverable erroneous situation from what I can gather, so I think returning nil rather than a `Result` type makes more sense here

 ```suggestion
    return NIL_VAL;
```